enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/HighAvailabilityMemberStateMachineTest.java

```diff
@@ -463,6 +504,171 @@ public Object answer( InvocationOnMock invocation ) throws Throwable
 
     }
 
+    @Test
+    public void whenHAModeSwitcherSwitchesToSlaveTheOtherModeSwitcherDoNotGetTheOldMasterClient() throws Throwable
+    {
+        InstanceId me = new InstanceId( 1 );
+        StoreId storeId = new StoreId();
+        HighAvailabilityMemberContext context = mock( HighAvailabilityMemberContext.class );
```

Comment:
All the mocking is pretty much this particular test dependent, I don't think we'll ever be able to reuse it.

I agree that this test is basically unreadable, but I don't know how to fix it since the code at hand is not particularly test friendly, hence this mock-all-the-thing solution.